### PR TITLE
Share button

### LINKED ISSUE #157: add links to share new URL to (via twitter, facebook, email?)
now that on click, each tree is assigned a URL (and the URL  links back to the tree marker), we can add a share button

@isabelle-wagenvoord @theswerd 

cf - NYC version (top left part of the sidepanel) - we can't have "favorite" as an option right now
I don't think the design should look quite like this one, just sharing for reference

Screen Shot 2019-08-05 at 7 16 58 PM

### PR DESCRIPTION
**Pull request by theswerd**
This request has 3 new share buttons finished

Twitter, Facebook, and Email

edit: fixes #157

### CODE REVIEW

**Comment by theswerd** [comment]
@emillipede Please let me know what you think

**Comment by emillipede** [comment]
thank you @isabelle-wagenvoord and @theswerd! 

overall these additions look great. I'll suggest a just couple of small changes to the text. 

the facebook button still doesn't show up, I'm not sure where the problem originates. any idea what's up? how does it look when you test it out? for now, we could either remove the facebook part from this branch and merge it or wait until the facebook part is resolved. 

the conflict in sidebar.js seems minor. I'll flag it so that you can double check.

thanks again to you both

File: public/js/Sidebar.js [hunk 1]
```diff
@@ -114,9 +118,23 @@ var app = this.app || {};
...
     this.defaultScreen.classList.add('hidden');
     this.treeContainer.classList.add('hidden');
-	this.vacantContainer.classList.add('hidden');
+  this.vacantContainer.classList.add('hidden');
     this.errorScreen.classList.remove('hidden');
   }
+
+  function buildTweetLink(tree) {
+    var url = window.location;
+    var tweet = `${tree.name_common} on Santa Monica's Public Tree Map: ${url}`;;
```
Comment (emillipede) [inline public/js/Sidebar.js]: I'd like to add @santamonicacity to the end of the tweet (it's the city's official twitter handle)

File: public/js/Sidebar.js [hunk 2]
```diff
@@ -114,9 +118,23 @@ var app = this.app || {};
...
+    return "https://twitter.com/intent/tweet?text=" + tweet;
+  }
+  function buildEmail(tree) {
+    var url = window.location;
+    var email = `${tree.name_common} on Santa Monica's Public Tree Map: ${url}`;;
+    return `mailto:?subject=Check%20out%20Santa%20Monica\'s%20Public%20Tree%20Map&body=${email}`;;
```
Comment (emillipede) [inline public/js/Sidebar.js]: can the subject of the email match the email text? eg '${tree.name_common} on Santa Monica's Public Tree Map' or '${tree.name_common} in Santa Monica'

File: public/js/Sidebar.js [hunk 3]
```diff
@@ -114,9 +118,23 @@ var app = this.app || {};
   Sidebar.prototype.showError = function() {
     this.defaultScreen.classList.add('hidden');
     this.treeContainer.classList.add('hidden');
-	this.vacantContainer.classList.add('hidden');
```
Comment (emillipede) [inline public/js/Sidebar.js]: the merge conflict's here

**Comment by isabelle-wagenvoord** [comment]
the button I made before only works when I tested the html file on localhost:8000.  it does not work when you just open the html file. looking at my previous commit right now, the button still works when I use localhost:8000:
image
i can't get ben's new button to work but maybe i'm missing something. can you try my previous commit by running it on localhost?

**Comment by emillipede** [comment]
Thank you for this clarification! 

When I run Ben's (the up-to-date branch) on localhost:8000, I don't see the facebook button. I'm not so familiar with this way of looking at PRs, I'm trying to sort out how to go backwards to look at your commit @isabelle-wagenvoord

**Comment by theswerd** [comment]
@emillipede @isabelle-wagenvoord Running mine now it doesn't seem to vote.

**Comment by isabelle-wagenvoord** [comment]
@emillipede, I used git checkout [commit id] on the command line to view it. thanks!

**Comment by emillipede** [comment]
thank you @isabelle-wagenvoord! I checked out that commit (18dd086dceb9955aa1ce2ce47427a4eb46c60fad) the way you suggest but I still don't see a button. the name in the sidebar is oddly aligned, too, for some species. I do see an error notice that the FB image file isn't found, could that be the reason the button isn't visible?

Screen Shot 2019-08-06 at 6 37 23 PM

**Comment by isabelle-wagenvoord** [comment]
i'm not getting that error when i view the commit, and i don't find any place in the code that references FB.png. however, it turns out i can't get the facebook button to show up anymore on this commit either now.

**Comment by emillipede** [comment]
I tried a couple of iterations of the code from the facebook developers page, but couldn't get it to work. I can ask around on Monday or on the slack channel to see if anyone's used this feature from facebook. any ideas @ian-r-rose? I think Sarah might have some experience with it 

I also realized it makes sense to include these sharing features in the vacant site version of the sidebar

**Comment by emillipede** [comment]
@hollydo recommends using [addtoany](https://www.addtoany.com/)

**Comment by emillipede** [comment]
resolved by #165

**Pull request closed without merging**